beginner_source/onnx/onnx_diagnostics_tutorial.py

```diff
@@ -0,0 +1,181 @@
+# --------------------------------------------------
+#
+# Let's begin by exporting a simple model:
+#
+
+import torch
+
+
+class Model(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.linear = torch.nn.Linear(3, 3)
+
+    def forward(self, x, y):
+        return self.linear(x) + y
+
+
+input_x = torch.randn(3, 3)
+input_y = torch.randint(0, 10, (3, 3))
+model = Model().eval()
+
+export_output = torch.onnx.dynamo_export(model, input_x, input_y)
+export_output.save("model.onnx")
+export_output.save_diagnostics("log.sarif")
+
+######################################################################
+# The SARIF file is saved as "log.sarif" in the current directory.
+#
+
+######################################################################
+# Loading and exploring SARIF diagnostics in SARIF Viewer
+# -------------------------------------------------------
+#
+# SARIF logs (\*.sarif) can be loaded in various ways:
+#
+# - Simply open the file, and the SARIF Viewer extension will automatically process it.
+# - Use the `sarif.showPanel` command and then select `Open SARIF log`.
+# Let's open the SARIF file (log.sarif) we just generated.
+#
+# The SARIF Viewer provides comprehensive tools to navigate and understand the diagnostics.
+# The top panel groups diagnostics by rules, offering a snapshot of the diagnostic rule name,
+# severity, message, and its source code location.
+#
+# .. image:: ../_static/img/onnx/diagnostics_sarif_viewer_top_panel.png
+#
+# Filters can be adjusted from the top panel. For example, we can filter the diagnostics by severity level.
+# .. image:: ../_static/img/onnx/diagnostics_sarif_viewer_filter.png
+#
+# By selecting a diagnostic, you'll access detailed information in the bottom panel.
+# Here, you'll find the full diagnostic message and further insights into the diagnostic rule.
+# .. image:: ../_static/img/onnx/diagnostics_sarif_viewer_bottom_panel.png
+#
+# The detailed information includes:
+#
+# - **Complete Diagnostic Message:** This includes comprehensive details specific to the code section
+#   from which this diagnostic originated. Often, it will also show the function signature related to
+#   the diagnostic occurrence. The next section will discuss how to capture even more information by
+#   increasing the diagnostic verbosity.
+# - **Rule Description:** This elaborates on the diagnostic rule, providing context about the identified
+#   issue, insights into its origin, and potential solutions. For a deeper dive into each rule, refer to
+#   `ONNX Diagnostic Rules <https://pytorch.org/docs/main/onnx_dynamo.html#diagnosing-issues-with-sarif>`_.
+#
+
+######################################################################
+# Increasing diagnostic verbosity for in-depth debugging
+# ------------------------------------------------------
+#
+# For a more detailed diagnostic output, adjust the `verbosity_level` in the `ExportOptions`.
```

Comment:
It would be nice to define what is verbosity in this context
Does increasing verbosity mean that more diagnostic rules be triggered or the same rules will be triggered, but each occurence will have more detail?